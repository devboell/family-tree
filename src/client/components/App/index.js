import React from 'react'
import { Route } from 'react-router-dom'

import FamilyTreePage from 'containers/FamilyTreePage'
import EditorPage from 'containers/EditorPage'
import Header from 'components/Header'

const App = () => (
  <div>
    <Header />
    <Route path="/" exact component={FamilyTreePage} />
    <Route path="/editor" exact component={EditorPage} />
  </div>
)

export default App
