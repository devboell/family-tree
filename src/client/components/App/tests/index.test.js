import React from 'react'
import App from '../index'


describe('App component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  it('FamilyTreePage Route', () => {
    const wrapper = shallow(<App />)
    const routeProps = wrapper.find('Route').at(0).props()
    expect(routeProps.component.name).toBe('FamilyTreePage')
    expect(routeProps.path).toBe('/')
  })

  it('EditorPage Route', () => {
    const wrapper = shallow(<App />)
    const routeProps = wrapper.find('Route').at(1).props()
    expect(routeProps.component.name).toBe('EditorPage')
    expect(routeProps.path).toBe('/editor')
  })
})
