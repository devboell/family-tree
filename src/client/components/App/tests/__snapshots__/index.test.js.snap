// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders without crashing 1`] = `
<div>
  <Header />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/editor"
  />
</div>
`;
